### PR TITLE
Node helper with intersections.

### PR DESCRIPTION
**Pull request by ThePrimeagen**
Allows for filtering a list of found nodes from a query with a row/col
intersection.

Let me know if there is anything you want changed.  I don't really know the style guides or the name style conventions for this library.

### CODE REVIEW

File: lua/nvim-treesitter/ts_utils.lua [hunk 1]
```diff
@@ -156,6 +156,38 @@ function M.get_root_for_position(line, col, root_lang_tree)
   return nil, nil, lang_tree
 end
 
+local function box_intersect(row, col, sRow, sCol, eRow, eCol)
```
Comment (theHamsta) [inline lua/nvim-treesitter/ts_utils.lua:159]: in general_we_use_sake_case in_this_repo

File: lua/nvim-treesitter/ts_utils.lua [hunk 2]
```diff
@@ -156,6 +156,38 @@ function M.get_root_for_position(line, col, root_lang_tree)
...
+-- Filters node list based on intersection with row and column
+--
+-- @param nodes the list of treesitter nodes
+-- @param row intersecting y column
+-- @param col intersecting x column
+function M.intersect_nodes(nodes, row, col)
+  local found = {}
+  for _, node in ipairs(nodes) do
+    local sRow, sCol, eRow, eCol = node:range()
+    if box_intersect(row, col, sRow + 1, sCol + 1, eRow + 1, eCol + 1) then
```
Comment (theHamsta) [inline lua/nvim-treesitter/ts_utils.lua:184]: how is this differren from `is_in_node_range`?

File: lua/nvim-treesitter/ts_utils.lua (hunk 1 above)
Comment (ThePrimeagen) [reply lua/nvim-treesitter/ts_utils.lua:159]: OHHH snap!  I didn't even recognize the fact that those weren't snake_case.  Haha.  It pains me to use it and I tried to conform as much as possible.   :)  Sometimes these old eyes don't see nothing.  I'll update this pr shortly.

File: lua/nvim-treesitter/ts_utils.lua (hunk 2 above)
Comment (ThePrimeagen) [reply lua/nvim-treesitter/ts_utils.lua:184]: Ohh nice!  This is great.  I don't have to have box_intersect.  I was making this for myself and don't quite know all the available functions.

**Comment by ThePrimeagen** [comment]
if there is anything else, let me know.  Also, if this isn't quite useful enough for tree sitter, let me know.

File: lua/nvim-treesitter/ts_utils.lua [hunk 3]
```diff
@@ -156,6 +156,21 @@ function M.get_root_for_position(line, col, root_lang_tree)
   return nil, nil, lang_tree
 end
 
+-- Filters node list based on intersection with row and column
+--
+-- @param nodes the list of treesitter nodes
+-- @param row intersecting y column
+-- @param col intersecting x column
```
Comment (vigoux) [inline lua/nvim-treesitter/ts_utils.lua:163]: I think we use three dashes for doc comments here :
```suggestion
--- Filters node list based on intersection with row and column
---
--- @param nodes the list of treesitter nodes
--- @param row intersecting y column
--- @param col intersecting x column
```

**Review by vigoux — APPROVED** [review]
Great, just a minor comment on style, otherwise it's good to go.

File: lua/nvim-treesitter/ts_utils.lua [hunk 4]
```diff
@@ -156,6 +156,21 @@ function M.get_root_for_position(line, col, root_lang_tree)
...
+function M.intersect_nodes(nodes, row, col)
+  local found = {}
+  for _, node in ipairs(nodes) do
+    if M.is_in_node_range(node, row, col) then
+      table.insert(found, node)
+    end
+  end
```
Comment (theHamsta) [inline lua/nvim-treesitter/ts_utils.lua:170]: Thanks for this nice addition!

**Review by theHamsta — APPROVED** [review]

**Comment by ThePrimeagen** [comment]
Please don't merge this. I've been thinking about it for some time. I think providing a `filter_nodes` function could be a lot better.

That way it's not just my change but other changes. And we can provide a couple simple helper functions to filter with.

**Comment by theHamsta** [comment]
I don't know whether this function is really necessary. I think it can already be done with `vim.tbl_filter(funtion(node) M.node_contains(node, row, col) end, nodes)`

**Comment by ThePrimeagen** [comment]
This sounds great. I don't want anything extra :) 

Some times I don't know the things available

**Pull request closed without merging**